pdns/credentials.cc

```diff
@@ -156,3 +160,89 @@ bool CredentialsHolder::isHashingAvailable()
   return false;
 #endif
 }
+
+#include <signal.h>
+#include <termios.h>
+
+std::string CredentialsHolder::readFromTerminal()
+{
+  struct termios term;
+  struct termios oterm;
+  memset(&term, 0, sizeof(term));
```

Comment:
WHy init `term` and `oterm` if you are going to overwrite them?

Reply:
They are not overwritten when we fall back to stdin, and we check the value of `term` no matter what. It seemed cleaner to initialize them both.